tests/unit/amo/components/TestSearchResult.js

```diff
@@ -131,19 +131,197 @@ describe(__filename, () => {
+          },
+        ],
+      }),
+    });
+
+    expect(root.find('.SearchResult-icon')).toHaveProp('alt', alt);
+  });
+
+  it('renders an empty string for the image alt tag while there is no addon', () => {
+    const root = render({
+      addon: {},
+    });
+
+    expect(root.find('.SearchResult-icon')).not.toHaveProp('alt', '');
+  });
+
+  it('displays the thumbnail image as the default src for static theme', () => {
+    const headerImageThumb =
+      'https://addons.cdn.mozilla.net/mytestthumb/12345.png';
+
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        type: ADDON_TYPE_STATIC_THEME,
+        previews: [
+          {
+            ...fakePreview,
+            thumbnail_url: headerImageThumb,
+          },
+        ],
+      }),
+    });
+    const image = root.find('.SearchResult-icon');
+
+    expect(image.prop('src')).toEqual(headerImageThumb);
+  });
+
+  it('displays srcSet values if preview has multiple options', () => {
+    const headerImageThumb = 'https://addons.cdn.mozilla.net/thumb/12345.png';
+    const headerImageFull = 'https://addons.cdn.mozilla.net/full/54321.png';
+    const thumbWidth = 450;
+    const fullWidth = 900;
+
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        type: ADDON_TYPE_STATIC_THEME,
+        previews: [
+          {
+            ...fakePreview,
+            image_url: 'https://addons.cdn.mozilla.net/thumb/notused.png',
+            thumbnail_url: 'https://addons.cdn.mozilla.net/thumb/notused.png',
+          },
+          {
+            ...fakePreview,
+            image_url: headerImageFull,
+            thumbnail_url: headerImageThumb,
+            image_size: [fullWidth, 200],
+            thumbnail_size: [thumbWidth, 100],
+          },
+        ],
+      }),
+    });
+    const image = root.find('.SearchResult-icon');
+
+    expect(image.prop('srcSet')).toEqual(
+      `${headerImageThumb} ${thumbWidth}w, ${headerImageFull} ${fullWidth}w`,
+    );
+  });
+
+  it('renders image without srcSet if there are no preview image sizes', () => {
+    const headerImageThumb = 'https://addons.cdn.mozilla.net/thumb/12345.png';
+    const headerImageFull = 'https://addons.cdn.mozilla.net/full/54321.png';
+
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        type: ADDON_TYPE_STATIC_THEME,
+        previews: [
+          {
+            ...fakePreview,
+          },
+          {
+            ...fakePreview,
+            image_url: headerImageFull,
+            thumbnail_url: headerImageThumb,
+            image_size: [],
+            thumbnail_size: [],
+          },
+        ],
+      }),
+    });
+    const image = root.find('.SearchResult-icon');
+    expect(image.prop('src')).toEqual(headerImageThumb);
+    expect(image).not.toHaveProp('srcSet');
+  });
+
+  it('renders image without srcSet if there is no large preview image url', () => {
```

Comment:
There are 3 different things that are checked which could result in no `srcSet` if any are false: no image URL, no `imageSize` or no `thumbSize`, so I think you could combine the two tests you have here into 1, which tests each of those cases. You could use `it.each()` for this. See https://github.com/mozilla/addons-frontend/blob/master/tests/unit/amo/components/TestHome.js#L67 for an example.